docs/reference/migration/migrate_2_0.asciidoc

```diff
@@ -408,6 +408,9 @@ In addition some query builders have been removed or renamed:
 * `filtered(...)` removed. Use `filteredQuery(...)` instead.
 * `inQuery(...)` removed.
 
+`TermsLookupQueryBuilder` has been removed, please use `TermsQueryBuilder` with no values and set the corresponding
```

Comment:
I removed it for the reason exposed previously, and documented the change in migration doc. 

> 3) TermsLookupQueryBuilder is a pure delegate to TermsQueryBuilder, and Queries.termsLookupQuery directly creates a new TermsQueryBuilder.

We can surely make that change in master first and rebased if you'd like.